engine/access/rest/websockets/controller.go

```diff
@@ -4,100 +4,225 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"sync"
+	"time"
 
 	"github.com/google/uuid"
 	"github.com/gorilla/websocket"
 	"github.com/rs/zerolog"
 
 	dp "github.com/onflow/flow-go/engine/access/rest/websockets/data_provider"
 	"github.com/onflow/flow-go/engine/access/rest/websockets/models"
-	"github.com/onflow/flow-go/engine/access/state_stream"
-	"github.com/onflow/flow-go/engine/access/state_stream/backend"
 	"github.com/onflow/flow-go/utils/concurrentmap"
 )
 
+const (
+	// PingPeriod defines the interval at which ping messages are sent to the client.
+	// This value must be less than pongWait.
+	PingPeriod = (PongWait * 9) / 10
+
+	// PongWait specifies the maximum time to wait for a pong message from the peer.
+	PongWait = 10 * time.Second
+
+	// WriteWait specifies the maximum duration allowed to write a message to the peer.
```

Comment:
I guess Ulyana took it from here https://github.com/gorilla/websocket/blob/v1.5.3/examples/chat/client.go#L23.

I believe it's because ping and pong share the same timer.

Let’s consider a case where `pongWait` is smaller than `pingPeriod`, and we’ll see why this configuration is problematic.

Parameters:
pongWait = 30s
pingPeriod = 40s

At t=0: 
The server sends a ping message to the client.

At t=30s:
The pongWait expires because the server hasn't received a pong (or any message) from the client.
The server assumes the connection is dead and closes it.

At t=40s:
The server sends its second ping, but the connection is already closed due to the timeout at t=30s.

Reply:
[3cfe98b](https://github.com/onflow/flow-go/pull/6757/commits/3cfe98b6271eee46981689fe2b6fe96913939bef)